outbox/order-service/src/main/java/io/debezium/examples/outbox/order/outbox/EventSender.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.examples.outbox.order.outbox;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.event.TransactionPhase;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import io.debezium.examples.outbox.order.event.ExportedEvent;
+
+@ApplicationScoped
+public class EventSender {
+
+    @Inject
+    private Event<OutboxEventSubmitted> submitted;
+
+    @Inject
+    private HouseKeeper houseKeeper;
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    public void onExportedEvent(@Observes ExportedEvent event) {
+        OutboxEvent outboxEvent = new OutboxEvent(
+                event.getAggregateType(), event.getAggregateId(), event.getType(), event.getPayload()
+        );
+        entityManager.persist(outboxEvent);
+
+        submitted.fire(new OutboxEventSubmitted(outboxEvent.getId()));
```

Comment:
Have you tried `INSERT` and `DELETE` in the same transaction? I belive it should still create two events and the table will be kept empty and code will be simplified.

Reply:
Good question, too. So Hibernate itself would optimize that and just do nothing. But what actually could work is to do this: insert - flush - delete. I'll try that.

Reply:
Ok, so I changed that.

> So Hibernate itself would optimize that and just do nothing

Turns out I was actually just dreaming that :) When doing `persist()` + `remove()` right after each other, it will produce an INSERT followed by DELETE. Steve confirmed there's no optimization for that. Next thing I wondered whether those could be omitted by the DB in the TX log, but that's not the case either. So it's all much simpler now indeed!